CMakeLists.txt

```diff
@@ -88,6 +88,9 @@ add_compile_options(-Wall -Wpedantic -Wextra)
 # This is needed everywhere so that we can catch segmentation faults
 add_compile_options(-fnon-call-exceptions)
 
+# We use C++17
+add_compile_options(-std=c++17)
```

Comment:
GCC defaults to C++17, but clang defaults to C++14. We use a few C++17 features a decent bit - e.g. std::filesystem. I think this is reasonable.